src/coreclr/jit/emitxarch.cpp

```diff
@@ -11461,6 +11466,9 @@ BYTE* emitter::emitOutputRR(BYTE* dst, instrDesc* id)
     }
     else if ((ins == INS_movsx) || (ins == INS_movzx) || (insIsCMOV(ins)))
     {
+        assert(hasCodeRM(ins) && !hasCodeMI(ins) && !hasCodeMR(ins));
+        code = insCodeRM(ins);
```

Comment:
Can you confirm why we added `insCodeRM()` and `AddVexPrefixIfNeeded()` here and below?

Reply:
Was it missing currently?

Reply:
we had logic like:
```
    // Get the 'base' opcode	
    code = insCodeRM(ins);	
    code = AddVexPrefixIfNeeded(ins, code, size);
    if ()
    { 
        code = insEncodeRMreg(ins, code); // us the existing value.
    } 
    else if ()
    {
       code = insCodeMR(ins);	//rewriting the existing value.
    }
```
so it was hard to follow, so I moved this 
```
    code = insCodeRM(ins);	
    code = AddVexPrefixIfNeeded(ins, code, size);
``` 
under conditions where it is used.